fe/fe-core/src/main/java/org/apache/doris/qe/QeProcessorImpl.java

```diff
@@ -206,6 +230,15 @@ public Map<String, QueryStatisticsItem> getQueryStatistics() {
 
     @Override
     public TReportExecStatusResult reportExecStatus(TReportExecStatusParams params, TNetworkAddress beAddr) {
+        if (params.isSetQueryProfile()) {
+            if (params.isSetBackendId()) {
+                Backend backend = Env.getCurrentSystemInfo().getBackend(params.getBackendId());
+                if (backend != null) {
+                    processQueryProfile(params.getQueryProfile(), backend.getHeartbeatAddress());
```

Comment:
Why not return directly after ```processQueryProfile``` if this is a reportProfileReq

Reply:
It's not a assumption, it clearly that only reportProfileRequest has queryProfile field, so after ```processQueryProfile```, it should return and not go forward, such as exec ```updateFragmentExecStatus```

Reply:
In the future, req of reportExecStatus could contain profile and execstatus at same time.